applications/OptimizationApplication/tests/test_sigmoidal_projection.py

```diff
@@ -0,0 +1,138 @@
+from KratosMultiphysics.KratosUnittest import TestCase
+
+# Import KratosUnittest
+import KratosMultiphysics.KratosUnittest as kratos_unittest
+
+class TestSigmoidalProjection(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.model = Kratos.Model()
+
+        cls.test_model_part =  cls.model.CreateModelPart("test")
+        cls.test_model_part.ProcessInfo.SetValue(Kratos.DOMAIN_SIZE, 3)
+
+        cls.test_model_part.CreateNewNode(1, 0.0,0.0,0.0)
+
+    def _GetContainerExpression(self):
+        return Kratos.Expression.NodalExpression(self.test_model_part)
+
+    def test_ProjectForward(self):
+        test_field = self._GetContainerExpression()
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1],[2],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[3,3],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,1],[2,3],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[1],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1],[3,4],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[2,1],[3,4],25.0,1)
+        with self.assertRaises(RuntimeError):
+            _ = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[4,3],25.0,1)
+
+        Kratos.Expression.LiteralExpressionIO.SetData(test_field, 1.5)
+        forward_projected_field = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[3,4],25.0,1)
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(forward_projected_field), 3.5, 4)
+
+        Kratos.Expression.LiteralExpressionIO.SetData(test_field, 5.0)
+        forward_projected_field = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[3,4],25.0,1)
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(forward_projected_field), 4.0, 4)
+
+        Kratos.Expression.LiteralExpressionIO.SetData(test_field, -10.0)
+        forward_projected_field = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[3,4],25.0,1)
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(forward_projected_field), 3.0, 4)
+
+        Kratos.Expression.LiteralExpressionIO.SetData(test_field, 1.51)
+        forward_projected_field = KratosOA.ControlUtils.SigmoidalProjectionUtils.ProjectForward(test_field,[1,2],[3,4],25.0,1)
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(forward_projected_field), 3.6224593312018545, 4)
```

Comment:
I would tighten the tolerances if possible just be safe. We had lot of cases these loosen tolerances backfiring ata later time.
```suggestion
        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(forward_projected_field), 3.6224593312018545, 10)
```